#!/usr/bin/env python
PACKAGE = "franka_panda_ip_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

"""
    Joint Position Controller (position_joint_position_controller) smoothing parameter. Increasing the value gives higher smoothing (smaller change in target position per controller update step).
"""
gen.add("target_filter_joint_pos_delta", double_t, 0, "Filter parameter that dictates how smoothly subsequent joint position commands are smoothed. Also responsible for speed of joint position execution.", 0.1, 0.0, 3.0) # (default, min, max)

gen.add("filter_joint_pos_delta", double_t, 0, "Filter joint pos.", 0.3, 0.0, 10.0) # (default, min, max)
gen.add("filter_factor_delta", double_t, 0, "Filter factor.", 0.01, 0.0, 3.0) # (default, min, max)
gen.add("param_change_filter_delta", double_t, 0, "Filter param change.", 0.005, 0.0, 3.0) # (default, min, max)

exit(gen.generate(PACKAGE, "controller_configurations", "joint_position_controller_ip_params"))
